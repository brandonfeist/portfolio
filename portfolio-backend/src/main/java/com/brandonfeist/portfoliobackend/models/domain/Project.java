package com.brandonfeist.portfoliobackend.models.domain;

import java.util.Date;
import java.util.Set;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

import lombok.Data;

@Data
@Entity
@Table(name = "projects")
public class Project {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @NotNull
  @Column(name = "image_url", nullable = false)
  private String imageUrl;

  @NotNull
  @Column(nullable = false)
  private String name;

  @NotNull
  @Column(nullable = false)
  private String summary;

  @NotNull
  @Column(nullable = false)
  private String description;

  @NotNull
  @ManyToMany(fetch = FetchType.EAGER)
  @JoinTable(
      name = "projects_technologies",
      joinColumns = { @JoinColumn(name = "project_id") },
      inverseJoinColumns = { @JoinColumn(name = "technology_id") }
  )
  private Set<Technology> technologies;

  @NotNull
  @Column(name = "project_date", nullable = false)
  private Date projectDate;

  @NotNull
  @Column(name = "created_date", nullable = false)
  private Date createdDate;

  @NotNull
  @Column(name = "updated_date", nullable = false)
  private Date updatedDate;
}
