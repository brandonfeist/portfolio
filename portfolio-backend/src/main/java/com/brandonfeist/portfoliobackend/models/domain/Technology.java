package com.brandonfeist.portfoliobackend.models.domain;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.Data;

@Data
@Entity
@Table(name = "technologies")
public class Technology {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(nullable = false)
  private String name;
}
