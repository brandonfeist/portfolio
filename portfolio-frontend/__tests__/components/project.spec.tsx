import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import Project from '../../components/project';
import Technology from '../../models/Technology';

describe('Component', () => {
  describe('Project', () => {
    it('renders correctly', function() {
      const projectDate: Date = new Date();
      const technologies: Array<Technology> = [{ name: 'React' }];
      const project = renderer
        .create(
          <Project
            name="Test"
            imageUrl="image.url"
            projectType="Personal"
            technologies={technologies}
            projectDate={projectDate}
          />,
        )
        .toJSON();

      expect(project).toMatchSnapshot();
    });
    it('should render Project props correctly', function() {
      // Things to render: name, type, tech, summary, image url, project date
    });
    it('should render without throwing an error', function() {
      const projectDate: Date = new Date();
      const technologies: Array<Technology> = [{ name: 'React' }];
      const wrap = mount(
        <Project
          name="Test"
          imageUrl="image.url"
          projectType="Personal"
          technologies={technologies}
          projectDate={projectDate}
        />,
      );
      expect(wrap.find('div').text()).toBe('TestPersonal - 2019');
    });
  });
});
