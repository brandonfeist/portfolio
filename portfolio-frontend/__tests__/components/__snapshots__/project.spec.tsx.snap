// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Project renders correctly 1`] = `
<div>
  <h1>
    Test
  </h1>
  <h2>
    Personal
     - 
    2019
  </h2>
  <img
    alt="Preview of Test"
    src="image.url"
  />
</div>
`;
